layers/headers.go

```diff
@@ -14,25 +14,26 @@ type XHeadersLayer struct {
 	XHeaders map[string]string
 }
 
-func (h *XHeadersLayer) OnRequest(state *httransform.LayerState) error {
+func (h *XHeadersLayer) OnRequest(ctx *layers.Context) error {
 	for k, v := range h.XHeaders {
-		state.RequestHeaders.SetString(k, v)
+		ctx.RequestHeaders.Append(k, v)
```

Comment:
Append adds new headers to the end of the list. I think, what you really need it [Set](https://pkg.go.dev/github.com/9seconds/httransform/v2@v2.0.5/headers#Headers.Set) here (as it was before).

Reply:
Thanks